app/controllers/takeovers/OtherBusinessAddressController.scala

```diff
@@ -95,27 +95,35 @@ class OtherBusinessAddressController @Inject()(val authConnector: PlayAuthConnec
     ctAuthorised {
-      registered { regId =>
-        val optBusinessName: Option[String] = request.session.get(businessNameKey)
-        val optAddressSeq: Option[Seq[NewAddress]] = request.session.get(addressSeqKey).map(Json.parse(_).as[Seq[NewAddress]])
-        (optBusinessName, optAddressSeq) match {
-          case (Some(businessName), Some(addressSeq)) => OtherBusinessAddressForm.form(businessName, addressSeq.length).bindFromRequest.fold(
-            formWithErrors =>
-              Future.successful(BadRequest(OtherBusinessAddress(formWithErrors, businessName, addressSeq))),
-            {
-              case OtherAddress =>
-                Future.successful(Redirect(regRoutes.AccountingDatesController.show())) //TODO redirect to ALF
-              case PreselectedAddress(index) =>
-                takeoverService.updateBusinessAddress(regId, addressSeq(index)).map {
-                  _ =>
-                    Redirect(regRoutes.AccountingDatesController.show()) //TODO redirect to next page when it's done
-                      .removingFromSession(businessNameKey)
-                      .removingFromSession(addressSeqKey)
-                }
-            }
-          )
-          case _ => Future.successful(Redirect(routes.OtherBusinessAddressController.show()))
-        }
+      registered {
+        regId =>
+          takeoverService.getTakeoverDetails(regId).flatMap {
+            optTakeoverDetails =>
+              val optAddressSeq: Option[Seq[NewAddress]] = request.session.get(addressSeqKey).map(Json.parse(_).as[Seq[NewAddress]])
+              (optTakeoverDetails, optAddressSeq) match {
+                case (Some(TakeoverDetails(_, Some(businessName), _, _, _)), Some(addressSeq)) => OtherBusinessAddressForm.form(businessName, addressSeq.length).bindFromRequest.fold(
+                  formWithErrors =>
+                    Future.successful(BadRequest(OtherBusinessAddress(formWithErrors, businessName, addressSeq))),
+                  {
+                    case OtherAddress =>
+                      addressLookupFrontendService.initialiseAlfJourney(
+                        handbackLocation = controllers.takeovers.routes.OtherBusinessAddressController.handbackFromALF(),
+                        specificJourneyKey = takeoversKey,
+                        lookupPageHeading = messagesApi("page.addressLookup.takeovers.otherBusinessAddress.lookup.heading", businessName),
+                        confirmPageHeading = messagesApi("page.addressLookup.takeovers.otherBusinessAddress.confirm.description", businessName)
+                      ).map(Redirect(_))
+                    case PreselectedAddress(index) =>
+                      takeoverService.updateBusinessAddress(regId, addressSeq(index)).map {
+                        _ =>
+                          Redirect(regRoutes.AccountingDatesController.show()) //TODO redirect to next page when it's done
+                            .removingFromSession(businessNameKey)
```

Comment:
any reference to businessName in session (in this class and its spec) should be removed since the submit doesn't pull it from there anymore